RevenueCatUI/PaywallView.swift

```diff
@@ -33,19 +33,20 @@ public struct PaywallView: View {
     @Environment(\.locale)
     private var locale
 
-    @StateObject
-    private var purchaseHandler: PurchaseHandler
+    let purchaseHandler: PurchaseHandler
 
-    @StateObject
-    private var introEligibility: TrialOrIntroEligibilityChecker
+    let introEligibility: TrialOrIntroEligibilityChecker
```

Comment:
Hmmm the reasoning was that this class doesn't observe any changes but I see that it passes it down to subview that do, so that needs to be changed back. I also see that a subview redeclares `introEligibility` as @StateObject again, when it should be using @ObservedObject at this point, so I'll go over this and make sure it's all correct. Thank you!

Reply:
This is now resolved. `purchaseHandler` has a couple of different possible owners, the `PaywallView` or the `PresentingPaywallModifier`. For this reason the `PaywallView` methods reference the a new property `@ObservedObject var purchaseHandler: PurchaseHandler`, where the object is either owned by the `PaywallView`'s `@StateObject var purchaseHandler: PurchaseHandler` OR the `PresentingPaywallModifier`'s `@StateObject var purchaseHandler: PurchaseHandler`. If `PaywallView` doesn't own the handler, then the @StateObject is set to a dummy handler that should never be used.